biojava-structure/src/main/java/org/biojava/nbio/structure/basepairs/BasePairParameters.java

```diff
@@ -0,0 +1,718 @@
+     * @return An integer value, number of base pairs
+     */
+    public int getLength() {
+        if (structure == null || pairParameters == null) throw new IllegalArgumentException();
+        return pairingParameters.length;
+    }
+
+
+    /**
+     * This reports all the pair parameters, in the order of:
+     * buckle, propeller, opening (in degrees), shear, stagger, stretch (in Å).
+     * @return A double[][] with length equal to number of base pairs for rows, and 6 columns
+     */
+    public double[][] getPairingParameters() {
+        return pairingParameters;
+    }
+
+    /**
+     * This reports all the base-pair step parameters, in the order of:
+     * tilt, roll, twist (in degrees), shift, slide, rise (in Å).
+     * @return A double[][] with length equal to number of base pairs (the first row 0 has no step
+     *  and therefore is six zeroes), and 6 columns.
+     */
+    public double[][] getStepParameters() {
+        return stepParameters;
+    }
+
+
+    /**
+     * This returns the primary strand's sequence where parameters were found.
+     * There are spaces in the string anywhere there was a break in the helix or when
+     * it goes from one helix to another helix in the structure. (the "step" is still returned)
+     * @return String of primary sequence with spaces between gaps and new helices.
+     */
+    public String getPairSequence() {
+        return pairSequence;
+    }
+
+
+    /**
+     * This returns the names of the pairs in terms of A, G, T/U, and C for each base pair group in the
+     * list.  The first character is the leading strand base and the second character is the complementary base
+     * @return
+     */
+    public List<String> getPairingNames() {
+        return pairingNames;
+    }
+
+    public List<Matrix4d> getReferenceFrames() {
+        return referenceFrames;
+    }
+
+
+    /**
+     * Return the buckle in degrees for the given base pair
+     * @param bp the number of the base pair (starting with 0)
+     * @return the value as a double (in degrees)
+     */
+    public Double getBuckle(int bp) {
+        if (bp < 0 || bp >= getPairingParameters().length) throw new IllegalArgumentException();
+        return pairingParameters[bp][0];
+    }
+
+    /**
+     * Return the propeller ("propeller-twist") in degrees for the given base pair
+     * @param bp the number of the base pair (starting with 0)
+     * @return the value as a double (in degrees)
+     */
+    public Double getPropeller(int bp) {
+        if (bp < 0 || bp >= getPairingParameters().length) throw new IllegalArgumentException();
+        return pairingParameters[bp][1];
+    }
+
+    /**
+     * Return the opening in degrees for the given base pair
+     * @param bp the number of the base pair (starting with 0)
+     * @return the value as a double (in degrees)
+     */
+    public Double getOpening(int bp) {
+        if (bp < 0 || bp >= getPairingParameters().length) throw new IllegalArgumentException();
```

Comment:
This check should be refactored into a method instead of repeated several times.  Or if guava is on the classpath for this module, use `checkArgument`